test/e2e/framework/resource/runtimeobj.go

```diff
@@ -55,6 +58,12 @@ func GetRuntimeObjectForKind(c clientset.Interface, kind schema.GroupKind, ns, n
 		return c.AppsV1().DaemonSets(ns).Get(context.TODO(), name, metav1.GetOptions{})
 	case kindBatchJob:
 		return c.BatchV1().Jobs(ns).Get(context.TODO(), name, metav1.GetOptions{})
+	case kindCustomCRD:
```

Comment:
hard-coding support for a single CRD-based type into this utility method doesn't make sense to me, and if we were going to, this should return the actual object (which would be an *unstructured.Unstructured object, not an autoscaling/v1 Scale object)